src/simulation/decision_model_simulation_results.jl

```diff
@@ -336,12 +290,12 @@ function _read_results(
     end
     existing_keys = list_result_keys(res, first(result_keys))
     _validate_keys(existing_keys, result_keys)
-    cached_results = get_cached_results(res, first(result_keys))
+    cached_results = get_cached_results(res, typeof(first(result_keys)))
```

Comment:
```suggestion
    cached_results = get_cached_results(res, eltype(result_keys))
```

Reply:
This change can be used all over

Reply:
Sadly it cannot so easily because lots of existing code passes in arrays of `OptimizationContainerKey` even though they contain only one concrete type, but I'll work on finding a more elegant solution

Reply:
Done